src/main/java/org/elasticsearch/index/query/CommonTermsQueryBuilder.java

```diff
@@ -217,4 +292,136 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     public String queryId() {
         return NAME;
     }
+
+    @Override
+    public Query toQuery(QueryParseContext parseContext) throws QueryParsingException, IOException {
+        String field;
+        FieldMapper mapper = parseContext.fieldMapper(fieldName);
+        if (mapper != null) {
+            field = mapper.names().indexName();
+        } else {
+            field = fieldName;
+        }
+
+        Analyzer analyzerObj;
+        if (analyzer == null) {
+            if (mapper != null) {
+                analyzerObj = parseContext.getSearchAnalyzer(mapper);
+            } else {
+                analyzerObj = parseContext.mapperService().searchAnalyzer();
+            }
```

Comment:
The last if only makes sense if the mapper is null, because otherwise it is already taken care of by `parseContext.getSearchAnalyzer`, that is return `mapperService().searchAnalyzer()` if the one at the field cannot be found. So I think the code behaves the same.